### ISSUE TITLE
Regular expression syntax error

### ISSUE THREAD

**Issue by januwA** [jviereck/regjsparser#105]
Hi, this regular expression parsing error

```js
const exp = /\s*(?<typedef>typedef)?\s*(?<struct>struct)\s*(?<structName>\w+)\s*{(?<props>[^}]*)}(\s*(?<aliasName1>\w+)?\s*,\s*(?<aliasName2>\*\w+)?\s*;\s*)?/gi;
```

https://github.com/babel/babel-loader/issues/888

[REPL](https://babel.dev/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=true&code_lz=GYVwdgxgLglg9mABANwBQEpEG8BQjEQIDOUiApgB4AOiAvIgPQA6RAVKgPwA8UAnlWQAmZYAD4-A4cHQcW7biQBOIaKKUqo6OZy7roAOQCGAWzKimAdwDUWtlh1VFcKkVEBtAHoBfALqt0Xqja3IYANjCGREamAIzm1jJyADTBXGERUSZkAEzmrJY2smwA3HIyDADmMMV4hGBEcKFkAHShcBWolFToNV44OGjoQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Ces2017%2Ctypescript&prettier=true&targets=&version=7.10.4&externalPlugins=)

**Comment by nicolo-ribaudo** [comment]
A smaller example:
```js
/{(?<x>)}/
```

**Comment by jviereck** [comment]
An even smaller rejected example is `/{}/.exec("{}") -> ["{}"]`. The RegExp `/{}/` seems to be valid in the browser but rejected by the parser.

My best guess is this is a special rule in the AnnexB of the spec, but I don't know. I will need to dig into this.

**Comment by nicolo-ribaudo** [comment]
I _think_ it is because of `Term` > `ExtendedAtom` > `ExtendedPatternCharacter`. We probably need to backtrack after entering `InvalidBracedQuantifier` (which always results in an error) if we can't finish parsing it.

**Comment by jviereck** [comment]
> I think it is because of `Term` > `ExtendedAtom` > `ExtendedPatternCharacter`

That looks reasonable. Thanks for digging this up.

The current `Term`s parsing is [here](https://github.com/jviereck/regjsparser/blob/gh-pages/parser.js#L484). I haven't figured out yet how to shape the code around to make this work.

**Comment by jviereck** [comment]
This error should be fixed by #106. There is a new version of regjsparser (0.6.5 and 0.6.6). I also published to NPM but at the point of writing it's not showing up on the webpage yet (though I received an email notification for the update).

**Comment by nicolo-ribaudo** [comment]
Npm is slow, sometimes it takes up to 15 mins to show the new version
